torchx/lib/torchx.ex

```diff
@@ -1,7 +1,84 @@
+defmodule Torchx.Macro do
+  defmacro __using__(_opts) do
+    quote do
+      import unquote(__MODULE__)
+      Module.register_attribute(Torchx, :torch_function, accumulate: true)
+    end
+  end
+
+  defmacro deftorch(call) do
+    {name, args} = Macro.decompose_call(call)
+
+    args_names = Enum.map(args, &elem(&1, 0))
+
+    if Enum.any?(args, &match?({:device, _, nil}, &1)) do
```

Comment:
Instead of using checking for the device variable, I would say it is cleaner to have two macros: maybe `deftorch` and `defdevice` or similar?